includes/attendee/attendee-repository.php

```diff
@@ -114,6 +115,43 @@ public function get_attendees( int $event_id ): array { // phpcs:ignore Generic.
 		return array();
 	}
 
+	/**
+	 * Get the hosts' users for an event.
+	 *
+	 * @param int $event_id The id of the event.
+	 * @return array[WP_User] The hosts of the event.
+	 */
+	public function get_hosts( int $event_id ): array {
+		global $wpdb, $gp_table_prefix;
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery
+		// phpcs:disable WordPress.DB.DirectDatabaseQuery.NoCaching
+		$host_ids = $wpdb->get_col(
+			$wpdb->prepare(
+				"
+				select user_id
+				from {$gp_table_prefix}event_attendees
+				where event_id = %d and is_host = 1
+			",
+				array(
+					$event_id,
+				)
+			)
+		);
+		// phpcs:enable
+
+		$event    = get_post( $event_id );
+		$host_ids = array_unique( array_merge( $host_ids, array( $event->post_author ) ) );
```

Comment:
The post author is [automatically added as a host](https://github.com/WordPress/wporg-gp-translation-events/blob/trunk/wporg-gp-translation-events.php#L220) when the event is published, so I don't think we need to do this here.

(Actually, currently it's not the post_author who is being added as a host, but the user who publishes the event, but as discussed previously, we should fix that).

Reply:
The problem with the author was resolved [here](https://github.com/WordPress/wporg-gp-translation-events/pull/200/files#diff-93ac378e918721e6698187d0123e47b161f8fe273064813e9cdef3433a9165a8L215).

If the event creator left the event, and we don't have another host, we will have a list of empty hosts. And if the `get_hosts` should return an array of `Attendee`, the creator won't appear there. So we need an array of `WP_User`, not of `Attendee`. What do you think the best place to put this method? Return it to the `Stats_Calculator`?

Reply:
> The problem with the author was resolved [here](https://github.com/WordPress/wporg-gp-translation-events/pull/200/files#diff-93ac378e918721e6698187d0123e47b161f8fe273064813e9cdef3433a9165a8L215).

Nice! I had missed that.

Reply:
The way I see it is that, now that we're tracking who is a host in the attendees table, the author of the event no longer has any relation to who is a host. The author is simply the user who created the event, and the hosts are the entries in the attendees table with `is_host` set to true.

To me this means that if there are no entries in the attendees table with `is_host` set to `true`, then the `get_hosts()` method should return an empty array.

However, as you mention, it should not be possible to have an event with no hosts. In the same way that we add the post_author as host when the event is published (to make sure the event has a host), we would also need to prevent the last host from un-attending the event.

This would mean that it's guaranteed that an event has at least one host, so the `get_hosts()` method will never return an empty array.

To me the solution would be to add check in the attend/un-attend code that prevents the last host from un-attending. Do you think this would make sense?

Reply:
Yes, I like this approach. We can disable the "You're attending" button, with a small text saying "You are the unique host, so you can't leave the event. Please, add a new host before". I will work on this approach.

Reply:
Isn't that also solved by the fact that the host cannot modify themselves, thus remove themselves? Thus hosts should not be allowed to unattend an event unless someone else releaves them from being a host.

Reply:
I don't think we should make this just for the last host but for any host. If you're a host, you cannot leave the event.

Reply:
The host can't modify herself, but she can stop attending to the event, so she will remove the row in the `attendees` table.

I think we could have situations where one event creator or a host wants to leave the event, and I think we should all her to do this, except she is the last one. One situation:
- A user creates an event that she will not attend: a physical event, an event in a different locale,...
- She adds a new host.
- She left the event.

For me, the most flexible approach is to allow leaving the event, expect the user is the last host, and avoid using the event creator to determine the hosts.